### PR TITLE
Ychoy/issue 44

### PR DESCRIPTION
**Pull request by ychoy**
- tests and methods for matrix_create and matrix_flip

### CODE REVIEW

File: algorithms/libs/matrix.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/*
+Creates a randomly generated matrix of single digit hexadecimal
+characters (as strings) by passing x and y coordinates of the matrix.
+*/
+const matrix_create = (x,y) => {
+  var result = [];
+  var x_coord = 0;
+  var y_coord = 0;
+  while (x_coord < x && y_coord < y) {
+    result.push([get_random_character(), get_random_character()]);
```
Comment (zpallin) [inline algorithms/libs/matrix.js]: If I am not mistaken, this looks like it only generates a 2 by X sized matrix... I don't think this is what you want for this function. It needs two loops, the outside loop bounded by X and the inside loop bounded by Y. For reach index X of result, you will generate a new array inside the first loop and `push` to it a random character for each iteration of Y.

File: algorithms/spec/MatrixSpec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    result = [["1","4","7"],["2","5","8"],["3","6","9"]];
+    expect(matrix.matrix_transpose(arr)).toEqual(result);
+  });
+  it('checks if [["1", "2", "3", "4"], ["5","6","7","8"], ["9", "10", "11", "12"], ["13", "14", "15", "16"]] is [["1","5","9", "13"], ["2","6","10", "14"], ["3","7","11", "15"],["4","8","12","16"]] transposed', () => {
+    arr = [["1", "2", "3", "4"], ["5","6","7","8"], ["9", "10", "11", "12"], ["13", "14", "15", "16"]];
+    result = [["1","5","9", "13"], ["2","6","10", "14"], ["3","7","11", "15"],["4","8","12","16"]];
+    expect(matrix.matrix_transpose(arr)).toEqual(result);
+  });
+
+  it('checks if [["1", "2", "3"], ["4","5","6"]] is [["1","4"],["2","5"],["3","6"]] transposed', () => {
```
Comment (zpallin) [inline algorithms/spec/MatrixSpec.js]: If this works, add one last test for the opposite transposition from `[["1","4"],["2","5"],["3","6"]]` to `[["1", "2", "3"], ["4","5","6"]]`.

**Comment by zpallin** [comment]
```
1. (null, [0, 1])
  next.map(0, 0) // val , idx
  prev[0] || [] -> [] .concat next[i] -> [0]
  next.map(1, 1)
  prev[1] || [] -> [] .concat next[i] -> [1]

  [[0],[1]]

2. ([[0],[1]], [2, 3])
  next.map(2, 0)
  prev[0] || [] -> [0] concat next[0] -> [0, 2]
  next.map(3, 1)
  prev[1] || [] -> [1] concat next[1] -> [1, 3]

3. ([[0, 2], [1, 3]], [4, 5])
  next.map(4, 0)
  prev[0] || [] -> [0, 2] concat next[0] -> [0, 2, 4]
  next.map(5, 1)
  prev[1] || [] -> [1, 3] concat next[1] -> [1, 3, 5]

  [[0, 2, 4], [1, 3, 5]]
```

File: algorithms/spec/MatrixSpec.js (hunk 2 above)
Comment (ychoy) [reply algorithms/spec/MatrixSpec.js]: Done

**Comment by ychoy** [comment]
@zpallin please review this PR

**Pull request merged**